lib/rspec/core/bisect/runner.rb

```diff
@@ -8,11 +8,18 @@ module Bisect
       # the given bisect server to capture the results.
       # @private
       class Runner
-        attr_reader :original_cli_args
+        attr_reader :original_cli_args, :original_spec_opts, :spec_opts
 
         def initialize(server, original_cli_args)
           @server            = server
           @original_cli_args = original_cli_args.reject { |arg| arg.start_with?("--bisect") }
+          @original_spec_opts = ENV['SPEC_OPTS']
+          if original_spec_opts =~ /--bisect/
+            bisect_arg = Shellwords.split(ENV["SPEC_OPTS"]).flatten.grep(/--bisect/).shift
```

Comment:
* IMO, using `shift` here is very odd.  `shift` is like `first` (a query) combined with `drop(1)` (a command) -- but you are only using it for the query part (since you keep no reference to the array being mutated), so I'd prefer to see `first` be used.
* Is `flatten` necessary?  I don't see how `Shellwords.split` can result in an array that is not already flattened.
* Using `grep(...).shift` (or `grep(...).first`) causes the entire list to be traversed, even if `--bisect` is one of the first arguments.  Instead, I'd use `find { ... }`.
* You are basically looking for `--bisect` in the spec opts multiple times: once in the conditional on the line above, and again on this line. That's wasted work.
* Actually, I'm not sure a conditional is even needed...

Putting all that together, I think this section can be simplified to:

``` ruby
ENV["SPEC_OPTS"] = Shellwords.split(ENV.fetch("SPEC_OPTS", "")).reject { |opt| opt =~ /--bisect/ }.shelljoin
```